packages/lib/sdk/src/usql/Query.js

```diff
@@ -443,15 +587,125 @@ export class Query {
 	/////////////////////
 	/// < Factories > ///
 	/////////////////////
-
+	/**
+	 * This is a fairly arbitrary number that determines how much data
+	 * the Query will cache internally. The larger the number, the
+	 * larger the cache will be.
+	 *
+	 * The number is based on our Query Score calculation, see
+	 * queryScore.js for details on how this is calculated.
+	 *
+	 * @default 5 * 10 * 1024
+	 *
+	 */
+	static CacheMaxScore = 5 * 10 * 1024;
+	/**
+	 * @type {Map<string, {added: number, query: Query<any>}>}
+	 */
 	static #cache = new Map();
 
+	static emptyCache = () => {
+		this.#cache.clear();
+	};
+
+	static get cacheSize() {
+		return this.#cache.size;
+	}
+
+	/**
+	 * @param {Query<any>} q
+	 */
+	static #addToCache = (q) => {
+		this.#cache.set(q.hash, {
+			query: q,
+			added: Date.now()
+		});
+
+		if (isDebug())
+			console.debug(`Added to cache: ${q.hash}`, {
+				cacheSize: this.#cache.size,
+				cacheScore: Array.from(this.#cache.values()).reduce((sum, q) => sum + q.query.score, 0)
+			});
+	};
+
+	/**
+	 * @template {QueryResultRow} [RowType=QueryResultRow]
+	 * @param {string} hash
+	 * @returns {Query<RowType> | null}
+	 */
+	static #getFromCache = (hash) => {
+		const cachedValue = this.#cache.get(hash);
+		if (cachedValue) {
+			return cachedValue.query;
+		}
+		return null;
+	};
+
+	static #cacheCleanup = () => {
+		let sumScore = Array.from(this.#cache.values()).reduce((sum, q) => sum + q.query.score, 0);
+		while (sumScore > this.CacheMaxScore) {
+			// TODO: Is this efficient enough? Sorting once per iteration is slightly gross
+			const oldest = Array.from(this.#cache.values()).sort((a, b) => a.added - b.added)[0];
+			this.#cache.delete(oldest.query.hash);
+			sumScore -= oldest.query.score;
+		}
+	};
+
+	/*
+		TODO: Write an update function?
+		Update will accept the same parameters as create (?)
+		It will return a promise race to load for 250ms and return the new store (or whenver the new store is loaded)
+		This will let the reactivity live outside of the Component
+	*/
+
+	/**
+	 * @param {import('./types.js').Runner} executeQuery
+	 * @param {string | import("@uwdata/mosaic-sql").Query} initialQuery
+	 * @param {import('./types.js').QueryOpts} [defaultOpts={}]
+	 * @param {number} [loadDelay=250]
+	 * @returns {{ initialValue: QueryValue, updater: <T extends QueryResultRow>(query: string, opts: import('./types.js').QueryOpts<T>) => Promise<QueryValue<T>>}}
+	 */
+	static reactive = (executeQuery, initialQuery, defaultOpts, loadDelay = 250) => {
+		/** @type {import('./types.js').CreateQuery<any>} */
+		const createFn = Query.create;
```

Comment:
✨ types ✨

Reply:
JSDoc doesn't allow you to make a generic function call if it can't infer the arguments, so this was the easiest way to get a ref to Query.create with an unscoped row type